.github/workflows/release.yml

```diff
@@ -39,9 +44,66 @@ jobs:
           python -m twine upload --username "${TWINE_USER}" --password "${TWINE_PASSWORD}" --skip-existing --repository-url ${REPO_URL} dist/*.whl
         env:
           REPO_URL: "https://upload.pypi.org/legacy/"
-      - if: always()
-        uses: elastic/apm-pipeline-library/.github/actions/notify-build-status@current
+
+  publish-lambda-layers:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: hashicorp/vault-action@v2.5.0
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/observability-team/ci/service-account/apm-agent-python access_key_id | AWS_ACCESS_KEY_ID ;
+            secret/observability-team/ci/service-account/apm-agent-python secret_access_key | AWS_SECRET_ACCESS_KEY
+      - name: Build lambda layer zip
+        run: ./dev-utils/make-distribution.sh
+      - name: Publish lambda layers to AWS
+        run: |
+          # Convert v1.2.3 to ver-1-2-3
+          VERSION=${GITHUB_REF_NAME/v/ver-}
+          VERSION=${VERSION//./-}
+        
+          ELASTIC_LAYER_NAME="elastic-apm-python-${VERSION}" .ci/publish-aws.sh
+      - uses: actions/upload-artifact@v3
+        with:
+          name: arn-file
+          path: ".arn-file.md"
+          if-no-files-found: error
+
+  github-draft:
+    needs:
+      - publish-lambda-layers
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: arn-file
+      - run: >- 
+          gh release create "${GITHUB_REF_NAME}"
+          --title="${GITHUB_REF_NAME}"
+          --generate-notes
+          --notes-file=".arn-file.md"
+          --draft
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+  notify:
+    runs-on: ubuntu-latest
+    if: always()
+    needs:
+      - publish-lambda-layers
+      - release
```

Comment:
What about `github-draft`? Though, it has a dependency with `publish-lambda-layers`, if added, will it work in the case `github-draft` failed?

Reply:
```suggestion
      - publish-pypi
      - github-draft
```